netdata: update to version 1.16.0

BKPepe opened the pull request:
Maintainers: me and @diizzyy 
Compile tested: Turris MOX, cortexa53, OpenWrt master
Run tested: Turris MOX, cortexa53, OpenWrt master

Description:

- Update to version [1.16.0](https://github.com/netdata/netdata/releases/tag/v1.16.0)
Add dependencies: libopenssl and liblz4

It was not necessary to refresh patches

Otherwise, compilations end up:

```
Package netdata is missing dependencies for the following libraries:
libcrypto.so.1.1
liblz4.so.1
libssl.so.1.1

```

Proof of run tested:
![Screenshot from 2019-05-27 23-39-39](https://user-images.githubusercontent.com/4096468/58440205-c4873f80-80d8-11e9-9a74-48d89b440dad.png)

![Screenshot from 2019-05-27 23-39-27](https://user-images.githubusercontent.com/4096468/58440204-c4873f80-80d8-11e9-8d77-f28fd2e07344.png)

diizzyy commented:
While the PR is fine I don't like of having OpenSSL and liblz4 as hard dependencies and I'd say it's a no go on 16Mbyte devices and lower. At least having mbedtls as an option would be much more preferable if SSL is a hard requirement.

It also looks like they're pulling in more deps in later versions :-/
https://github.com/netdata/netdata/blob/master/configure.ac#L348

We should also make sure the cloud stuff is disabled by default (not sure it is)

neheb commented:
Looks like there is a python-mbedtls: https://pypi.org/project/python-mbedtls/

No idea how easy it would be to integrate. OpenSSL is unfortunately king of the hill.

neheb commented:
Needs a rebase.

diizzyy commented:
If we want to bump this further I'd propose that we rename 1.14.0 to netdata-legacy and the newer versions for netdata or something along those lines.

diizzyy commented:
@BKPepe 
Ping?

BKPepe commented:
Rebased on current master. Updated to version 1.16.0. 
Run, tested Turris MOX, cortexa53, OpenWrt master.

According to the details from [OpenWrt Hamburg Meeting in 2019](https://openwrt.org/meetings/hamburg2019/start), the upcoming stable release (OpenWrt 19.07) will be the last release with support for 4/ 32 MB devices, so it is question if this is still blocker.

With the [cloud](https://blog.netdata.cloud/posts/netdata-cloud-announcement/), I'm fine as you need to Sign In to Netdata cloud.  Seems to be very useful in the end.

diizzyy commented:
I'd say yes on both 8Mbyte and 16Mbyte devices unfortunately.

BKPepe commented:
Do you want to have netdata-legacy here in this repository or in [packages-abandoned](https://github.com/openwrt/packages-abandoned)?

diizzyy commented:
It's still very usable so I think we should keep it in tree for now.

anomeome commented:
```
Package netdata is missing dependencies for the following libraries:
libjson-c.so.4
```
Is there a change to requirement of build platform?

BKPepe commented:
See https://github.com/openwrt/packages/pull/9695
